src/components/AddressSearch/index.js

```diff
@@ -382,6 +389,21 @@ function AddressSearch(props) {
                                 />
                             </View>
                         }
+                        renderRow={(data) => {
+                            const title = data.isPredefinedPlace ? data.name : data.structured_formatting.main_text;
+                            const subtitle = data.isPredefinedPlace ? data.description : data.structured_formatting.secondary_text;
```

Comment:
Let's set up the title and subtitle completely here before we return JSX, and add a comment explaining it. I understand it because I read your GH comment, but I wouldn't otherwise.

Something like:
```js
// For recent waypoints saved as predefined places, they might not have a name field since it was added after recent waypoints were already saved on the backend. If that's the case, fallback on the description, which is the address of the waypoint and will always be set.
```

Reply:
Please ignore this for now until we decide that this is in fact the route we want to take.

Reply:
Made the changes in accordance with the path decided on slack. Please let me know if the comment is still needed